### ISSUE TITLE
SINGULARITYENV_PATH not taking affect

### ISSUE THREAD

**Issue by robbinslewis** [apptainer/singularity#1821]
## Version of Singularity:

2.6.0

### Expected behavior

execute programs located in SINGULARITYENV_PATH

### Actual behavior

executables are not found from within the image

### Steps to reproduce behavior

I upgraded to 2.6.0 hoping this was related to #1430 but doesn't seem to have helped

$ export SINGULARITYENV_PATH=/opt/linuxbrew/bin
$ env | grep -i SINGULARITYENV
SINGULARITYENV_PATH=/opt/linuxbrew/bin

launch container and check for the path:
$ singularity exec /share/pkg/singularity/images/biobakery2017-12-13.img env|grep linuxbrew
SING_USER_DEFINED_PATH=/opt/linuxbrew/bin

try to run an executable within the path:
$ singularity exec /share/pkg/singularity/images/biobakery2017-12-13.img biobakery_demos --help
/.singularity.d/actions/exec: 9: exec: biobakery_demos: not found

verify the executable exists in the path:
$ singularity exec /share/pkg/singularity/images/biobakery2017-12-13.img /opt/linuxbrew/bin/biobakery_demos --version
biobakery_demos v0.0.1

**Comment by jmstover** [comment]
Instead of `SINGULARITYENV_PATH`, try using either:

`SINGULARITYENV_PREPEND_PATH`
`SINGULARITYENV_APPEND_PATH`

This does require a image created with an updated Singularity as well.

**Comment by robbinslewis** [comment]
I did try SINGULARITYENV_APPEND_PATH and was getting the same behavior.


The image was built with an older version of singularity, I'll try rebuilding it with 2.6.0.

Is there a way to convert an older image to a newer, similar to converting to a writable image or do I need to start fresh?


Lewis Robbins | Systems Admin Engineer Sr.
Data Sciences & Technology
Lewis.Robbins@umassmed.edu
O. 774-455-4353



Confidentiality Notice:

This e-mail message, including any attachments, is for the sole use of the intended recipient(s) and may contain confidential, proprietary and privileged information. Any unauthorized review, use, disclosure or distribution is prohibited. If you are not the intended recipient, please contact the sender immediately and destroy or permanently delete all copies of the original message.
________________________________

**Comment by jmstover** [comment]
There isn't a way to convert images. Where this is handled is in the files under `/.singularity.d/` in the image, and that is created by the specific singularity version.

**Comment by GodloveD** [comment]
Sorry, @jmstover is right.  There was a significant amount of discussion on this issues.  In the end, @gmkurtzer stepped in and said that if we messed with the contents of `/.singularity.d/env` at runtime (thereby fixing old containers) we would be introducing a kludge that we would need carry forward in future versions of Singularity.  So we only fixed the issue for containers built after the fix went in.
